src/v/cluster/partition.cc

```diff
@@ -67,6 +72,9 @@ partition::partition(
     if (is_id_allocator_topic(_raft->ntp())) {
         _id_allocator_stm = ss::make_shared<cluster::id_allocator_stm>(
           clusterlog, _raft.get());
+    } else if (is_tx_registry_topic(_raft->ntp())) {
```

Comment:
Would it be possible not to make this logic dependent from partition name, i know that we already have lots of stms handle in this way but maybe it would be good not to do it again as we will move the other out from the partition

Reply:
Partition and stms belong together because they share the life cycle: every time a partition starts the stms should start too and the same works for stopping. We may introduce a stm manager and let a partition host it but overall it leads almost to the same situation. Synced offline and decided to postpone this exercise until we decide to abstract stms to use with wasm to have user-defined stms.